TombEngine/Objects/TR3/Entity/tr3_tiger.cpp

```diff
@@ -82,117 +88,162 @@ namespace TEN::Entities::Creatures::TR3
-					else if (TestProbability(TIGER_WALK_CHANCE))
-						item->Animation.TargetState = TIGER_STATE_WALK_FORWARD;
-				}
-				else if (AI.bite && AI.distance < TIGER_BITE_ATTACK_RANGE)
-					item->Animation.TargetState = TIGER_STATE_BITE_ATTACK;
-				else if (AI.bite && AI.distance < TIGER_POUNCE_ATTACK_RANGE)
-				{
-					item->Animation.TargetState = TIGER_STATE_POUNCE_ATTACK;
-					creature->MaxTurn = TIGER_POUNCE_ATTACK_TURN_RATE_MAX;
-				}
-				else if (item->Animation.RequiredState)
-					item->Animation.TargetState = item->Animation.RequiredState;
-				else if (creature->Mood != MoodType::Attack && TestProbability(TIGER_ROAR_CHANCE))
-					item->Animation.TargetState = TIGER_STATE_ROAR;
-				else
-					item->Animation.TargetState = TIGER_STATE_RUN_FORWARD;
-
-				break;
-
-			case TIGER_STATE_WALK_FORWARD:
-				creature->MaxTurn = TIGER_WALK_TURN_RATE_MAX;
-
-				if (creature->Mood == MoodType::Escape ||
-					creature->Mood == MoodType::Attack)
-				{
-					item->Animation.TargetState = TIGER_STATE_RUN_FORWARD;
-				}
-				else if (TestProbability(TIGER_ROAR_CHANCE))
-				{
-					item->Animation.TargetState = TIGER_STATE_IDLE;
-					item->Animation.RequiredState = TIGER_STATE_ROAR;
-				}
-
-				break;
-
-			case TIGER_STATE_RUN_FORWARD:
-				creature->MaxTurn = TIGER_RUN_TURN_RATE_MAX;
-
-				if (creature->Mood == MoodType::Bored)
-					item->Animation.TargetState = TIGER_STATE_IDLE;
-				else if (creature->Flags && AI.ahead)
-					item->Animation.TargetState = TIGER_STATE_IDLE;
-				else if (AI.bite && AI.distance < TIGER_RUN_ATTACK_RANGE)
-				{
-					if (LaraItem->Animation.Velocity.z == 0.0f)
+					}
+					else if (creature->Mood == MoodType::Escape && Lara.TargetEntity != item && ai.ahead)
+					{
 						item->Animation.TargetState = TIGER_STATE_IDLE;
-					else
-						item->Animation.TargetState = TIGER_STATE_RUN_SWIPE_ATTACK;
-				}
-				else if (creature->Mood != MoodType::Attack && TestProbability(TIGER_ROAR_CHANCE))
-				{
-					item->Animation.TargetState = TIGER_STATE_IDLE;
-					item->Animation.RequiredState = TIGER_STATE_ROAR;
-				}
-				else if (creature->Mood == MoodType::Escape &&
-					Lara.TargetEntity != item && AI.ahead)
-				{
-					item->Animation.TargetState = TIGER_STATE_IDLE;
-				}
-
-				creature->Flags = 0;
-				break;
-
-			case TIGER_STATE_BITE_ATTACK:
-			case TIGER_STATE_RUN_SWIPE_ATTACK:
-			case TIGER_STATE_POUNCE_ATTACK:
-				if (!creature->Flags && item->TouchBits.Test(TigerAttackJoints))
-				{
-					DoDamage(creature->Enemy, TIGER_ATTACK_DAMAGE);
-					CreatureEffect(item, TigerBite, DoBloodSplat);
-					creature->Flags = 1; // 1 = is attacking.
-				}
-
-				break;
+					}
+
+					creature->Flags = 0;
+					break;
+
+				case TIGER_STATE_BITE_ATTACK:
+				case TIGER_STATE_POUNCE_ATTACK:
+					if (!creature->Flags && item->TouchBits.Test(TigerBiteAttackJoints))
+					{
+						if	(item->Animation.AnimNumber == GetAnimNumber(*item, TIGER_ANIM_BITE_ATTACK) &&
```

Comment:
I would add extra pair of brackets here before `||` operator, because eventually compiler may warn about logical operation precedence.